src/contract.ts

```diff
@@ -42,7 +42,22 @@ export async function handle(
 ): Promise<ContractReadResult | ContractWriteResult> {
   const input = action.input;
 
-  // tick state on any interaction, even when reading, so users get the most recent evaluation
+  // don't tick on evolutions, it should only update the source code transaction
+  if (input.function === 'evolve') {
+    return evolve(state, action);
+  }
+
+  // TODO: this is an interaction specific for testing and updating state without having to fork the contract, it should be removed for mainnet deployment
+  if (input.function === 'evolveState') {
+    return evolveState(state, action);
+  }
+
+  // directly call tick for tick interaction
+  if (input.function === 'tick') {
+    return tick(state);
```

Comment:
Can this just be a case where we do nothing in the switch? Also should be be awaiting here?

Reply:
we don't need to await - the contract returns a Promise so it will be awaited in the evaluation. 

and sure, we can just return `tickedState` in the switch as oppose to pulling it up